src/App.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react';
+import {
+  View,
+  StyleSheet,
+  Text,
+  Platform,
+  ActivityIndicator,
+} from 'react-native';
+import NewsDetails from './screens/NewsDetails';
+import NewsList from './screens/NewsList';
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isNewsSelected: false,
+      isNewsAvailable: false,
+      isLoading: false,
+      article,
+    };
+    this.handleNewsSelected.bind(this);
+    this.handleDetailsClose.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchNews();
+  }
+
+  handleNewsSelected(article) {
+    this.setState({ isNewsSelected: true, article });
+  }
+
+  handleDetailsClose() {
+    this.setState({ isNewsSelected: false });
+  }
+
+  async fetchNews() {
+    try {
+      this.setState({ isLoading: true });
+      const response = await fetch(
```

Comment:
You could extract this into api/news.js into function `fetchNews()` and then just call it here.